darkman.el

```diff
@@ -109,14 +110,26 @@ when the mode is changed."
   "Get a theme from the ‘darkman-themes’ which corresponds to the current mode."
   (darkman--lookup-theme (darkman-get)))
 
-(defun darkman--mode-changed-signal-handler (new-mode)
-  "Handle the ModeChanged signal.  NEW-MODE is the new mode."
+(defun darkman-load-theme-mode (new-mode)
+  "Load a theme associated with the NEW-MODE."
   (let ((new-theme (darkman--lookup-theme new-mode)))
     (unless darkman-switch-themes-silently
       (message (format "Darkman switched to %s mode, switching to %s theme."
-		       new-mode new-theme)))
+                       new-mode new-theme)))
     (load-theme new-theme)))
 
+(defun darkman--call-event-handler (interface property value)
+  "Handle method_call events on Darkman DBus.
+
+INTERFACE is the name of the interface that is the target of the event.
+
+PROPERTY is the property that is modified by the event.
+
+VALUE is the new value of PROPERTY."
+  (when (and (string-equal "Mode" property)
+             (equal darkman--dbus-service interface))
+    (darkman-load-theme-mode (car value))))
```

Comment:
I would personally like to see the handler logic remain in the handler function, this split-up is a bit confusing given the scope of the handler is relatively simple.

I'm thinking of something like this:

```emacs-lisp
(defun darkman--call-event-handler (interface property value)
  "Handle method_call events on Darkman DBus.
INTERFACE is the name of the interface that is the target of the event.
PROPERTY is the property that is modified by the event.
VALUE is the new value of PROPERTY."
  (when (and (string-equal "Mode" property)
             (equal darkman--dbus-service interface))
    (let* ((new-mode (car value))
	   (new-theme (darkman--lookup-theme new-mode)))
      (unless darkman-switch-themes-silently
	(message (format "Darkman switched to %s mode, switching to %s theme."
			 new-mode new-theme)))
      (load-theme new-theme))))
```

Reply:
Yes, I noticed that it might be too much indirection in case we only have 1-3 cases. Let me change it.

Reply:
Thanks!